src/audio/mixin_mixout/mixin_mixout_hifi3.c

```diff
@@ -22,84 +22,49 @@ static void normal_mix_channel_s16(struct audio_stream __sparse_cache *sink, int
 				   int32_t frame_count, uint16_t gain)
 {
 	int frames_to_mix, frames_to_copy, left_frames;
-	int n, nmax, i, m, left;
+	int n, nmax, i, m;
 	ae_int16x4 in_sample;
 	ae_int16x4 out_sample;
-	ae_int16x4 *in;
-	ae_int16x4 *out;
-	ae_valign inu = AE_ZALIGN64();
-	ae_valign outu1 = AE_ZALIGN64();
-	ae_valign outu2 = AE_ZALIGN64();
 	/* audio_stream_wrap() is required and is done below in a loop */
-	ae_int16 *dst = (ae_int16 *)sink->w_ptr + start_frame;
-	ae_int16 *src = (ae_int16 *)source->r_ptr;
+	ae_int16x4 *dst = (ae_int16x4 *)((ae_int16 *)sink->w_ptr + start_frame);
+	ae_int16x4 *src = (ae_int16x4 *)source->r_ptr;
 
 	assert(mixed_frames >= start_frame);
 	frames_to_mix = AE_MIN_32_signed(mixed_frames - start_frame, frame_count);
 	frames_to_copy = frame_count - frames_to_mix;
 	n = 0;
 
 	for (left_frames = frames_to_mix; left_frames > 0; left_frames -= n) {
-		src = audio_stream_wrap(source, src + n);
-		dst = audio_stream_wrap(sink, dst + n);
+		src = audio_stream_wrap(source, src);
+		dst = audio_stream_wrap(sink, dst);
```

Comment:
this change looks suspicious to me. Are `src` and `dst` changed in the loop? If not, why do they need to be wrapped?

Reply:
nice, thanks! **If** you make a new version maybe you could add a comment there for xtensa-extrinsic coders like myself :-) But not critical of course